frontend/src/api/RestaurantApi.tsx

```diff
@@ -49,3 +49,37 @@ export const useSearchRestaurants = (
 
   return { results, isLoading };
 };
+
+export const useGetRestaurant = (restaurantId: string) => {
+  const { getAccessTokenSilently } = useAuth0();
+
+  const createGetRestaurant = async (): Promise<Restaurant> => {
+    const accessToken = await getAccessTokenSilently();
+
+    const response = await fetch(
+      `${API_BASE_URL}/api/restaurant/${restaurantId}`,
+      {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+        },
+      }
+    );
+
+    if (!response.ok) {
+      throw new Error('Failed to get restaurant');
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/45#issuecomment-2413380674>#d153a0</a><span> - 10/15/2024, 09:47 am</span>
<br>

<div><small>🟠</small> Medium importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The error handling is too generic and doesn't provide specific information about the type of error encountered, which could make debugging and user feedback more difficult.
</div>
<div id="fix">
<b>Fix:</b> Enhance the error handling to provide more specific error messages based on the response status or any error information from the API.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -69,3 +69,7 @@
      if (!response.ok) {
 -      throw new Error('Failed to get restaurant');
 +      const errorMessage = await response.text();
 +      throw new Error(
 +        `Failed to get restaurant: ${response.status} ${response.statusText}. 
 +        ${errorMessage || 'No additional error information available.'}`
 +      );
      }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged